src/utils/geometry/filterPoints.ts

```diff
@@ -0,0 +1,56 @@
+import { Image } from '../..';
+
+import { Point } from './points';
+
+interface FilterPointsOptions {
+  /**
+   * The number of points that should be removed if they are close to extremum.
+   */
+  removeClosePoints: number;
+  /**Shows what kind of extremum is being computed */
+  kind: 'minimum' | 'maximum';
+}
+/**
+ * Finds extreme values of an image which are not stacked together.
+ * @param points - Array of points that should be combined to improve.
+ * @param image - Image which extrema are calculated from.
+ * @param options - FilterPointsOptions
+ * @returns Array of Points.
+ */
+export function filterPoints(
+  points: Point[],
+  image: Image,
+  options: FilterPointsOptions,
+) {
+  let { removeClosePoints = 0, kind = 'maximum' } = options;
+  const isMax = kind === 'maximum';
+
```

Comment:
You are using the first channel of the image and this is quite arbitrary.
Please allow an option 'channel'. If channel is unspecified and nb channels of image > 1, throw an error.